src/NUnitEngine/nunit.engine.tests/Services/TestRunnerFactoryTests/TestCases/NetStandardTestCases.cs

```diff
@@ -30,145 +30,177 @@ namespace NUnit.Engine.Tests.Services.TestRunnerFactoryTests.TestCases
+                SetName($"{{m}}({testName}");
+            }
+        }
+
         public static IEnumerable<TestCaseData> TestCases
         {
             get
             {
-                var testName = "Single assembly (string ctor)";
-                var package = TestPackageFactory.OneAssemblyStringCtor();
-                var expected = new RunnerResult { TestRunner = typeof(LocalTestRunner) };
-                yield return new TestCaseData(package, expected).SetName($"{{m}}({testName})");
+                yield return new TestRunnerFactoryData(
+                    "SingleAssembly (string ctor)",
+                    new TestPackage("a.dll"),
+                    new RunnerResult<LocalTestRunner>()
+                );
 
-                testName = "Single assembly (list ctor)";
-                package = TestPackageFactory.OneAssemblyListCtor();
-                expected = new RunnerResult { TestRunner = typeof(LocalTestRunner) };
-                yield return new TestCaseData(package, expected).SetName($"{{m}}({testName})");
+                yield return new TestRunnerFactoryData(
+                    "Single assembly (list ctor)",
+                    new TestPackage(new[] { "a.dll" }),
+                    new RunnerResult<LocalTestRunner>()
+                );
 
-                testName = "Two assemblies";
-                package = TestPackageFactory.TwoAssemblies();
-                expected = new RunnerResult
-                {
-                    TestRunner = typeof(AggregatingTestRunner),
-                    SubRunners = new[]
-                    {
-                        new RunnerResult { TestRunner = typeof(LocalTestRunner) },
-                        new RunnerResult { TestRunner = typeof(LocalTestRunner) }
-                    }
-                };
-                yield return new TestCaseData(package, expected).SetName($"{{m}}({testName})");
+                yield return new TestRunnerFactoryData(
+                    "Two assemblies",
+                    new TestPackage(new[] { "a.dll", "b.dll" }),
+                    new RunnerResult<AggregatingTestRunner, LocalTestRunner, LocalTestRunner>()
+                );
 
 #if NETCOREAPP2_0
-                testName = "Single project (list ctor)";
-                package = TestPackageFactory.OneProjectListCtor();
-                expected = new RunnerResult
-                {
-                    TestRunner = typeof(AggregatingTestRunner),
-                    SubRunners = new[]
-                    {
-                        new RunnerResult { TestRunner = typeof(LocalTestRunner) }
-                    }
-                };
-                yield return new TestCaseData(package, expected).SetName($"{{m}}({testName})");
+                yield return new TestRunnerFactoryData(
+                    "SingleProject (list ctor)",
+                    new TestPackage(new[] { "a.nunit" }),
+                    new RunnerResult<LocalTestRunner>()
+                );
```

Comment:
This test looks broken both before and after - I don't believe either the before or after TestRunner tree is correct here. Let's handle this in a separate issue.

Reply:
I do think the after is correct, however, since LocalTestRunner can aggregate any number of assembies found in the project.